opal/mca/common/ofi/common_ofi.c

```diff
@@ -25,3 +26,291 @@ int mca_common_ofi_register_mca_variables(void)
         return OPAL_ERROR;
     }
 }
+
+/* check that the tx attributes match */
+static int
+check_tx_attr(struct fi_tx_attr *provider_info,
+              struct fi_tx_attr *provider)
+{
+    if (!(provider->msg_order & ~(provider_info->msg_order)) &&
+        !(provider->op_flags & ~(provider_info->op_flags)) &&
+        (provider->inject_size == provider_info->inject_size)) {
+        return 0;
+    } else {
+        return OPAL_ERROR;
+    }
+}
+
+/* check that the rx attributes match */
+static int
+check_rx_attr(struct fi_rx_attr *provider_info,
+              struct fi_rx_attr *provider)
+{
+    if (!(provider->msg_order & ~(provider_info->msg_order)) &&
+        !(provider->op_flags & ~(provider_info->op_flags))) {
+        return 0;
+    } else {
+        return OPAL_ERROR;
+    }
+}
+
+/* check that the ep attributes match */
+static int
+check_ep_attr(struct fi_ep_attr *provider_info,
+              struct fi_ep_attr *provider)
+{
+    if (!(provider->type & ~(provider_info->type)) &&
+        !(provider->mem_tag_format & ~(provider_info->mem_tag_format)) &&
+        (provider->max_msg_size == provider_info->max_msg_size) &&
+        (provider->tx_ctx_cnt == provider_info->tx_ctx_cnt) &&
+        (provider->rx_ctx_cnt == provider_info->rx_ctx_cnt)) {
+        return 0;
+    } else {
+        return OPAL_ERROR;
+    }
+}
+
+/* check that the provider attributes match */
+static int
+check_provider_attr(struct fi_info *provider_info,
+                    struct fi_info *provider)
+{
+    /* make sure both info are the same provider and provide the same attributes */
+    if (0 == strcmp(provider_info->fabric_attr->prov_name, provider->fabric_attr->prov_name) &&
+        !check_tx_attr(provider_info->tx_attr, provider->tx_attr) &&
+        !check_rx_attr(provider_info->rx_attr, provider->rx_attr) &&
+        !check_ep_attr(provider_info->ep_attr, provider->ep_attr) &&
+        !(provider_info->caps & ~(provider->caps)) &&
+        !(provider_info->mode & ~(provider->mode))) {
+        return 0;
+    } else {
+        return OPAL_ERROR;
+    }
+}
+
+/* Takes a device BDF and returns it's NUMA node's logical index.
```

Comment:
TRIVIAL nit that may very well not be worth doing anything about -- typo in the comment. Should be "its".

Reply:
I'll fix this, thanks!